app/javascript/file_controllers/tree_controller.js

```diff
@@ -0,0 +1,234 @@
+          levelMatches.pop()
+          levelMatches.push(thisLevel-1)
+        } else {
+          rowElement.classList.add('search-hidden')
+        }
+      } 
+    })
+    this.restripe()
+    this.updateCount()
+  }
+
+  lastLevelMatch(levelMatches) {
+    return levelMatches[levelMatches.length - 1]
+  }
+
+  searchMatch(rowElement, search) {
+    return rowElement.dataset.filepath.includes(search) || rowElement.querySelector('*[data-tree-role="label"]')?.textContent.toLowerCase().includes(search)
+  }
+
+  clearSearch() {
+    this.tableTarget.querySelectorAll('tbody > tr').forEach((rowElement) => {
+      rowElement.classList.remove('search-hidden')
+    })
+    this.restripe()
+    this.updateCount()
+  }
+
+  childTrElements(trElement) {
+    const childTrElements = []
+    let nextTrElement = trElement.nextElementSibling
+    while(nextTrElement) {
+      childTrElements.push(nextTrElement)
+      nextTrElement = nextTrElement.nextElementSibling
+      if(nextTrElement && nextTrElement.getAttribute('aria-level') <= trElement.getAttribute('aria-level')) {
+        nextTrElement = null
+      }
+    }
+    return childTrElements
+  }
+
+  navigate(event) {
+    switch(true) {
+      case event.key === 'Enter' && this.isBranch(event.target):
+        event.preventDefault()
+        this.toggle(event)
+        break;
+      case event.key === 'Enter' && this.isLeaf(event.target):      
+        const downloadElement = this.downloadElement(event.target)
+        if(!downloadElement) break;
+
+        event.preventDefault()
+        downloadElement.click()
+        break;
+      case event.key === 'ArrowRight' && this.isClosedBranch(event.target):
+        event.preventDefault()
+        this.toggle(event)
+        break;
+      case event.key === 'ArrowRight' && this.isOpenBranch(event.target):
+        event.preventDefault()
+        event.target.nextElementSibling.focus()
+        break;
+      case event.key === 'ArrowLeft' && this.isOpenBranch(event.target):
+        event.preventDefault()
+        this.toggle(event)
+        break;
+      case event.key === 'ArrowLeft' && event.target.getAttribute('aria-level') > 1:
+        const parentBranchRowElement = this.parentBranchRowElement(event.target)
+        if(!parentBranchRowElement) break;
+
+        event.preventDefault()
+        parentBranchRowElement.focus()
+        break;
+      case event.key === 'ArrowUp':
+        const previousRowElement = this.previousBranchRowElement(event.target)
+        if(!previousRowElement) break;
+        
+        event.preventDefault()
+        previousRowElement.focus()
+        break;
+      case event.key === 'ArrowDown':
+        const nextRowElement = this.nextBranchRowElement(event.target)
+        if(!nextRowElement) break;
+        
+        event.preventDefault()
+        nextRowElement.focus()
+        break;
+      case event.key === 'Home':
+        event.preventDefault()
+        this.tableTarget.querySelector('tbody > tr').focus()
```

Comment:
Can you make this tr a stimulus target?

Reply:
I switched this to use `this.visibleRows` instead.